src/machine.rs

```diff
@@ -159,13 +159,20 @@ pub struct AllocExtra {
     /// Data race detection via the use of a vector-clock,
     ///  this is only added if it is enabled.
     pub data_race: Option<data_race::AllocExtra>,
+    /// Weak memory emulation via the use of store buffers,
+    ///  this is only added if it is enabled.
+    pub weak_memory: Option<weak_memory::AllocExtra>,
 }
 
 /// Extra global memory data
 #[derive(Debug)]
 pub struct MemoryExtra {
     pub stacked_borrows: Option<stacked_borrows::MemoryExtra>,
     pub data_race: Option<data_race::MemoryExtra>,
+    // Weak memory emulation shares global state with race detector,
+    // but we still need to know whether the user has opted into it
+    // when we initialise per-allocation metadata
+    weak_memory: bool,
```

Comment:
Why not store this bool inside data_race then?

Reply:
This is gone due to upstream Miri refactor (removing `Machine` I think?)

Reply:
Rather, removing `MemoryExtra` and putting that all into `Machine`.